pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_pantheon_yaml_file.adoc

```diff
@@ -0,0 +1,45 @@
+[id='con-pantheon-yaml-file_{context}']
+
+= The Pantheon configuration file
+
+In order to use a Git repository to import content to Pantheon, you need to create a configuration file in the root directory of your repository named `pantheon2.yml`. This yaml file indicates to Pantheon that it can interact with files in the repository.
+
+The configuration file enables the following:
+
+* Importing content from a Git repository into Pantheon
+* Uploading modules into Pantheon using the uploader tool
+
+The configuration file defines the following:
+
+`server`:: The URL of the Pantheon server.
+`repository`:: A unique ID to reference the URL of the repository. You specify the actual URL of the repository when you synchronize the repository the first time.
```

Comment:
I did not understand the difference between a unique ID and the actual URL.

Reply:
Yeah, I think an example would help. @benradey Can you please provide an example of a unique ID vs a URL for a repo, and how the unique ID and repo are associated with each other in a command-line?

Reply:
Sure.

The `repository` field is supposed to be a unique human-readable name that currently serves only one purpose - to ensure that the mapping between 'git repository' and 'pantheon repository' is persistent. (Otherwise, Pantheon currently has no memory of a repository that has been previously synced, vs. a brand-new repository being synced for the first time.) In the future, the repository name might also be used to achieve cross-repository module includes for assemblies, but details here haven't been ironed out yet.

The `server` field is used mostly for debugging right now - it's only used when invoking the pantheon uploader from the command line (keep in mind that command line usage is unsupported for MS2) so that the uploader knows where to attempt to sync the files to. What invoking a git sync from the Pantheon UI, the `server` field is ignored because the location of the Pantheon instance is obviously known.

Does that help?

Reply:
> keep in mind that command line usage is unsupported for MS2

Will it be supported in the future? In other words, should I document it or not?

Reply:
It will be supported in the future for uploading to personal sandboxes, which is a feature that does not exist today.

It might be helpful to document the usage of the tool for troubleshooting purposes. Users can't run the tool against our production instance of Pantheon, but they could run it against our lower environments. Perhaps somebody is trying to troubleshoot their module/resource wildcard patterns, and is running the tool against our stage environment to make sure that the correct set of files gets uploaded.

So bottom line, I believe that having it documented would be helpful.

Reply:
I think that for it to be useful in the way you describe, we should have a separate topic on Troubleshooting, where we say specifically how to do what you're referring to. Otherwise, I don't see it as helpful. Also, who would be doing the troubleshooting you're talking about? CCS? DPX? @adahms I'd like your input here.

Reply:
Thanks for the ping, @stoobie, and my apologies for the delay in getting back to you.

I think it's useful to have the content and procedures today, even if we don't necessarily include them in the official help section in the software right away. We can always expand on or refine them in the future as the need arises, and since it's a piece of functionality that does already exist, the content you've created can help general users understand what it's all about.

As for a troubleshooting procedure, I'm a little worried that could expand the scope of this particular issue beyond its original boundaries, so perhaps that's something we can raise a ticket for in the future when it's needed.